File: python/blog/sort/insertionSort.py

def insertionSort(arr):
    for i in range(1, len(arr)):
        tmp = arr[i];
        j = i - 1;
        
        while (j >= 0 and arr[j] > tmp):
            arr[j + 1] = arr[j]
            j -= 1
        
        arr[j + 1] = tmp
    
    return arr

print(insertionSort([5,2,1,8,3,3]));

File: c/blog/sort/insertionSort.c
#include <stdio.h>

void print(int* arr, int len)
{
	for (int i = 0; i < len; i++)
		printf("%d ", *arr++);
	printf("\n");
}

void insertionSort(int* arr, int len)
{
	int tmp, j;

	for(int i = 1; i < len; i++) {
		tmp = arr[i];
		j = i - 1;

		while(j >= 0 && arr[j] > tmp) {
			arr[j + 1] = arr[j];
			j--;
		}

		arr[j + 1] = tmp;
	}
}

void main()
{
	int arr[] = {5,2,1,8,3,3};

	printf("before sort: ");
	print(arr, 6);
	
	insertionSort(arr, 6);

	printf("after sort: ");
	print(arr, 6);
}
